zcash_client_sqlite/src/error.rs

```diff
@@ -72,6 +73,9 @@ pub enum SqliteClientError {
     /// The account for which information was requested does not belong to the wallet.
     AccountUnknown,
 
+    /// The account being added collides with an existing account in the wallet with the given ID.
+    AccountCollision(AccountId),
+
```

Comment:
I don't see where errors of this type are being created, and it doesn't make sense to me; account IDs are opaque autoincrementing values, there should never be a collision between them.

Reply:
This isn't about an account ID colliding. It's about the _account_ colliding, as detectable by (seed fingerprint, index) or viewing key collisions.

The `AccountId` provided as an argument _is_ the opaque account ID value, and is provided so that the caller can match the account up with one shown in the app, so the app can tell the user "oops, you already imported this account. It's _this one_.".

Reply:
I will add code that returns this error as part of adding tests that exercise the failure cases

Reply:
Can you add the code to detect this collision in the implementation of account insertion, and return it from there? My primary complaint was that the enum variant was added without it ever being created. If the check for collision were implemented, I would have been better able to understand its use. The comment here should also be expanded to describe under what conditions (fingerprint/zip32 index, viewing key conflict) the collisions can happen.

In the case of viewing key import, what happens if you attempt to import an FVK where you currently have an IVK for the same account? The reverse should be a collision, but does FVK import constitute an "upgrade" that is allowed to silently occur?

Reply:
I just pushed the first few tests, and actual use of the `AccountCollision` variant. I'll add another test or two later today.

As for IVK/FVK differences, I agree with you. But the table doesn't store IVKs at all at the moment, so when we add support for IVKs, we should include handling of IVK collisions and upgrades.